eclipse-collections/src/main/java/org/eclipse/collections/impl/set/immutable/AbstractImmutableSet.java

```diff
@@ -472,6 +473,14 @@ public boolean isProperSubsetOf(SetIterable<? extends T> candidateSuperset)
     @Override
     public ParallelUnsortedSetIterable<T> asParallel(ExecutorService executorService, int batchSize)
     {
-        return this.toSet().asParallel(executorService, batchSize);
+        if (executorService == null)
+        {
+            throw new NullPointerException();
+        }
+        if (batchSize < 1)
+        {
+            throw new IllegalArgumentException();
+        }
+        return new NonParallelUnsortedSetIterable<>(this);
```

Comment:
What's the reason for this change? It's non-parallel before and after?

Reply:
@motlin before it was in O(n), now it's O(1) because before it was calling toSet.